src/redux/doctor/doctorSlice.js

```diff
@@ -48,6 +67,9 @@ export const doctorSlice = createSlice({
       .addCase(fetchAllDoctors.rejected, (state, action) => {
         state.status = 'failed';
         state.error = action.payload;
+      })
+      .addCase(addNewDoctor, (state, action) => {
+        state.doctors.push(action.payload);
```

Comment:
You cannot mutate the redux state directly, only reassign it. 

`state.doctors = [...state.doctors, action.payload]`